### PR TITLE
Allows adding compression at method level

### PR DESCRIPTION
**Pull request by fedefernandez**
As per [this comment](https://github.com/grpc/grpc-java/issues/2196#issuecomment-242141285) seems that this is the right way for enabling compression on the server.

@juanpedromoreno thoughts?

### CODE REVIEW

File: modules/common/src/main/scala/protocol.scala [hunk 1]
```diff
@@ -32,7 +32,12 @@ sealed trait SerializationType extends Product with Serializable
 case object Protobuf           extends SerializationType
 case object Avro               extends SerializationType
 
-class rpc(val serializationType: SerializationType)                   extends StaticAnnotation
+sealed abstract class CompressionType extends Product with Serializable
+case object Identity                  extends CompressionType
```
Comment (juanpedromoreno) [inline modules/common/src/main/scala/protocol.scala]: I'm wondering if we could rename this by something like as `Plain` or similar, what do you think?

**Comment by juanpedromoreno** [comment]
Minor comment added, however, this looks great, thanks @fedefernandez !

File: modules/common/src/main/scala/protocol.scala (hunk 1 above)
Comment (fedefernandez) [reply modules/common/src/main/scala/protocol.scala]: Yeah, you know, I was trying to be cohesive with [grpc-java](https://github.com/grpc/grpc-java/blob/master/core/src/main/java/io/grpc/Codec.java) but I agree it's not the best name.

Comment (juanpedromoreno) [reply modules/common/src/main/scala/protocol.scala]: I missed that, forget my suggestion ;)

File: modules/internal/src/main/scala/server/monixCalls.scala [hunk 2]
```diff
@@ -34,41 +34,54 @@ object monixCalls {
...
-  def clientStreamingMethod[F[_]: Effect, Req, Res](f: Observable[Req] => F[Res])(
-      implicit S: Scheduler): ClientStreamingMethod[Req, Res] =
+  def clientStreamingMethod[F[_]: Effect, Req, Res](
+      f: Observable[Req] => F[Res],
+      maybeCompression: Option[String])(implicit S: Scheduler): ClientStreamingMethod[Req, Res] =
     new ClientStreamingMethod[Req, Res] {
 
-      override def invoke(responseObserver: StreamObserver[Res]): StreamObserver[Req] =
+      override def invoke(responseObserver: StreamObserver[Res]): StreamObserver[Req] = {
+        addCompression(responseObserver, maybeCompression)
```
Comment (fedefernandez) [inline modules/internal/src/main/scala/server/monixCalls.scala]: I'm not very cool with this approach, I was trying to change the `addConmpression` method result to `F[Unit]` but I'm not sure how to add an effect here, in the `transformStreamObserver` that needs to be applied before anything else. I think I'll give it a second try.

**Pull request merged**